nixos/lib/systemd-unit-options.nix

```diff
@@ -147,6 +156,20 @@ in rec {
         '';
       };
 
+      upholds = mkOption {
+        default = [];
+        type = types.listOf unitNameType;
+        description = lib.mdDoc ''
+          Configures dependencies similar to Wants=, but as long as this unit is up, all units
+          listed in Upholds= are started whenever found to be inactive or failed, and no job is
+          queued for them. While a Wants= dependency on another unit has a one-time effect when
+          this units started, a Upholds= dependency on it has a continuous effect, constantly
+          restarting the unit if necessary. This is an alternative to the Restart= setting of
+          service units, to ensure they are kept running whatever happens. The restart happens
+          without delay, and usual per-unit rate-limit applies.
+        '';
```

Comment:
I don't think copying the man page is a good idea. We don't do that for other systemd options, and it could easily get out of sync as systemd updates its documentation. It'd be better to just refer to the man page.

Reply:
Addressed, I think. Replaced the blurb with a shorter summary and I imagine if people are looking at these options, they can figure out to check the upstream document, just like they are able to for `wants` and `requires`.